chat-sdk/src/test/java/com/amazon/connect/chat/sdk/network/WebSocketManagerTest.kt

```diff
@@ -0,0 +1,140 @@
+package com.amazon.connect.chat.sdk.network
+
+import com.amazon.connect.chat.sdk.provider.ConnectionDetailsProvider
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.runTest
+import okhttp3.OkHttpClient
+import okhttp3.WebSocket
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import java.lang.reflect.Field
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class WebSocketManagerTest {
```

Comment:
I don't think we're covering many of the core cases in this test file.  We should really be testing connection/disconnection/reconnection logic here as well as `parseTranscriptItemFromJson`.

I would reference a file like [ChatServiceImplTest.kt](https://github.com/amazon-connect/amazon-connect-chat-android/blob/be10746e946871eb056f76fc018aa1ec11ba2261/chat-sdk/src/test/java/com/amazon/connect/chat/sdk/repository/ChatServiceImplTest.kt) for how we want to implement unit tests.